plugins/accounts-settings.c

```diff
@@ -0,0 +1,474 @@
+/*
+ *
+ *  oFono - Open Source Telephony
+ *
+ *  Copyright (C)  2016 Canonical Ltd.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <sys/types.h>
+#include <pwd.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include <glib.h>
+#include <gdbus.h>
+#include <systemd/sd-login.h>
+
+#include <ofono.h>
+
+#define OFONO_API_SUBJECT_TO_CHANGE
+#include <ofono/types.h>
+#include <ofono/log.h>
+#include <ofono/plugin.h>
+#include <ofono/system-settings.h>
+
+#define ACCOUNTS_SERVICE          "org.freedesktop.Accounts"
+#define ACCOUNTS_PATH             "/org/freedesktop/Accounts/User"
+#define ACCOUNTS_PHONE_INTERFACE  "com.ubuntu.touch.AccountsService.Phone"
+#define DBUS_PROPERTIES_INTERFACE "org.freedesktop.DBus.Properties"
+
+struct setting {
+	char* ofono_name;
+	char* property_name;
+};
+
+static const struct setting setting_names[] = {
+	{ PREFERRED_VOICE_MODEM, "DefaultSimForCalls" }
+};
+
+#define NUM_SETTINGS G_N_ELEMENTS(setting_names)
+
+struct accounts_data {
+	sd_login_monitor *login_monitor;
+	guint login_watch;
+	char *property_values[NUM_SETTINGS];
+	uid_t current_uid;
+	guint prop_change_watch;
+	char uid_path[sizeof(ACCOUNTS_PATH) + 32];
+};
+
+static struct accounts_data accounts_g;
```

Comment:
ofono convention uses a "g_" prefix for global variables ( see src/modem.c, or drivers/isimodem/uicc.c ).